### PR TITLE
Quote names containing comma or semicolon

### LINKED ISSUE #578: JSON quoted comma not ignored
#### Issue Summary

I'm using a python script to send emails. It mostly works.  Comma embedded in quotes are causing the submission to fail.

#### Steps to Reproduce

The input is a CSV file.

"Joe User",addr1@example.com,54321
"User, Joe",addr2@example.com,55555
The second line fails.

Python correctly parses the line and formats it as JSON, but it fails on submission. Apparently the quotes on name field are ignored and sees the embedded comma as a delimiter. It then uses User as the name and Joe as the email address and fails.

Is this a known bug? Is there a work-around?

#### Technical details:

* sendgrid-python 5.3.0
* Python Version: 2.7.3

### PR DESCRIPTION
**Pull request by cmccandless**
# Fixes #578 

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Escapes common CSV delimiters in Email.name

### CODE REVIEW

**Comment by thinkingserious** [comment]
@cmccandless,

Could you please merge this agains [this branch](https://github.com/sendgrid/sendgrid-python/tree/v4)? Thanks!

With Best Regards,

Elmer

**Comment by cmccandless** [comment]
There... finished the fight with the CI. This PR is ready to merge into v4.

**Comment by thinkingserious** [comment]
Hello @cmccandless,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

File: sendgrid/helpers/mail/email.py [hunk 1]
```diff
@@ -35,6 +49,14 @@ def name(self):
 
     @name.setter
     def name(self, value):
+        if not (value is None or isinstance(value, str)):
+            raise TypeError('name must be of type string.')
+
+        # Escape common CSV delimiters as workaround for
+        # https://github.com/sendgrid/sendgrid-python/issues/578
+        if value is not None and (',' in value or ';' in value):
+            value = html_entity_decode(value)
```
Comment (medmunds) [inline sendgrid/helpers/mail/email.py]: I think you actually want [`email.utils.quote`](https://docs.python.org/3/library/email.util.html#email.utils.quote) here. (Email display names don't really have anything to do with HTML entities.)

File: sendgrid/helpers/mail/email.py [hunk 2]
```diff
@@ -35,6 +49,14 @@ def name(self):
 
     @name.setter
     def name(self, value):
+        if not (value is None or isinstance(value, str)):
+            raise TypeError('name must be of type string.')
+
+        # Escape common CSV delimiters as workaround for
+        # https://github.com/sendgrid/sendgrid-python/issues/578
+        if value is not None and (',' in value or ';' in value):
```
Comment (medmunds) [inline sendgrid/helpers/mail/email.py]: Also `or '"' in value`. (`"` in the name will cause the same problems as comma.)

**Pull request merged**